systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -265,6 +266,33 @@ public static HelmClient helmClient() {
         return resourceReady[0];
     }
 
+    /**
+     * Auxiliary method for copying Controller {@link Constants.TEST_SUITE_NAME_LABEL} and {@link Constants.TEST_CASE_NAME_LABEL} labels
+     * into PodTemplate ensuring that in case of failure {@link io.strimzi.systemtest.logs.LogCollector} will collect all
+     * related Pods, which corespondents to such Controller (i.e., Job, Deployment)
+     *
+     * @param resource controller resource from which we copy test suite or test case labels
+     * @param resourcePodTemplate {@link PodTemplateSpec} of the specific resource
+     * @param <T> resource, which sings contract with {@link HasMetadata} interface
+     * @param <R> {@link PodTemplateSpec}
+     */
+    public <T extends HasMetadata, R extends PodTemplateSpec> void copyTsOrTcControllerLabelsIntoPodTemplate(T resource, R resourcePodTemplate) {
+        // 1. fetch Controller and PodTemplate labels
+        final Map<String, String> controllerLabels = resource.getMetadata().getLabels();
+        final Map<String, String> podLabels = resourcePodTemplate.getMetadata().getLabels();
```

Comment:
Is this safe with regards to no labels / unmodifiable map of labels?

Reply:
Now it is :)